File: Animation-OC/Animation-OC/RotationViewController.m

//
//  RotationViewController.m
//  Animation-OC
//
//  Created by chenyufeng on 15/12/31.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

#import "RotationViewController.h"

@interface RotationViewController ()

@property (weak, nonatomic) IBOutlet UIView *greenSquare;//旋转一次；

@property (weak, nonatomic) IBOutlet UIView *redSquare;//旋转无数次；


@end

@implementation RotationViewController

- (void)viewDidLoad {
    [super viewDidLoad];


}

- (void)viewWillAppear:(BOOL)animated{

  [super viewWillAppear:animated];


  [self spinGreenSquare];
  [self spinRedSquare];

  
}



- (void)spinGreenSquare{
  [UIView animateWithDuration:2 animations:^{

    self.greenSquare.transform = CGAffineTransformRotate(self.greenSquare.transform, M_PI);//一个PI，180度；
  } completion:^(BOOL finished) {

  }];

}

- (void)spinRedSquare{

  [UIView animateWithDuration:1 delay:0 options:UIViewAnimationOptionCurveLinear animations:^{
  self.redSquare.transform = CGAffineTransformRotate(self.redSquare.transform, 360);//一个PI，180度；
  } completion:^(BOOL finished) {
    [self spinRedSquare];
  }];
  
}

@end

File: PositionAnimation01/PositionAnimation01/RotationViewController.swift
//
//  RotationViewController.swift
//  PositionAnimation01
//
//  Created by chenyufeng on 15/8/20.
//  Copyright (c) 2015年 chenyufengweb. All rights reserved.
//


import UIKit

class RotationViewController: UIViewController {

    @IBOutlet weak var wheel: UIImageView!
  
    override func viewDidLoad() {

        super.viewDidLoad()
    }
    
    func spin(){

        UIView.animateWithDuration(1, delay: 0, options: .CurveLinear, animations: {

            self.wheel.transform = CGAffineTransformRotate(self.wheel.transform, CGFloat(360))//第二个参数是角度；
            }) {
                //结束后仍旧调用spin()函数；
                (finished)-> Void in
                self.spin()
        }
    }//spin()
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        
        self.spin()
    }
}












